include/seqan3/search/detail/search_common.hpp

```diff
@@ -23,13 +23,26 @@ namespace seqan3::detail
 struct search_param
 {
     //!\brief Total number of errors (upper bound over all error types).
-    uint8_t total;
+    uint8_t total{};
     //!\brief Total number of substitution errors.
-    uint8_t substitution;
+    uint8_t substitution{};
     //!\brief Total number of insertion errors.
-    uint8_t insertion;
+    uint8_t insertion{};
     //!\brief Total number of deletion errors.
-    uint8_t deletion;
+    uint8_t deletion{};
+
+    //!\brief Returns `true` if all member variables of `lhs` and `rhs` are equal, `false` otherwise.
+    constexpr friend bool operator==(search_param const & lhs, search_param const & rhs) noexcept
```

Comment:
test?

Reply:
it is tested in search_configuration_test where I compare the max_error configuration that holds a now `search_param` struct